wasm/prover/src/prover.rs

```diff
@@ -0,0 +1,482 @@
+
+#[derive(strum_macros::EnumMessage, Debug, Clone, Copy)]
+#[allow(dead_code)]
+enum ProverPhases {
+    #[strum(message = "Connect application server with websocket proxy")]
+    ConnectWsProxy,
+    #[strum(message = "Build prover config")]
+    BuildProverConfig,
+    #[strum(message = "Set up prover")]
+    SetUpProver,
+    #[strum(message = "Bind the prover to the server connection")]
+    BindProverToConnection,
+    #[strum(message = "Spawn the prover thread")]
+    SpawnProverThread,
+    #[strum(message = "Attach the hyper HTTP client to the TLS connection")]
+    AttachHttpClient,
+    #[strum(message = "Spawn the HTTP task to be run concurrently")]
+    SpawnHttpTask,
+    #[strum(message = "Build request")]
+    BuildRequest,
+    #[strum(message = "Start MPC-TLS connection with the server")]
+    StartMpcConnection,
+    #[strum(message = "Received response from the server")]
+    ReceivedResponse,
+    #[strum(message = "Parsing response from the server")]
+    ParseResponse,
+    #[strum(message = "Close the connection to the server")]
+    CloseConnection,
+    #[strum(message = "Start notarization")]
+    StartNotarization,
+    #[strum(message = "Commit to data")]
+    Commit,
+    #[strum(message = "Finalize")]
+    Finalize,
+    #[strum(message = "Notarization complete")]
+    NotarizationComplete,
+    #[strum(message = "Create Proof")]
+    CreateProof,
+}
+
+fn log_phase(phase: ProverPhases) {
+    log!(
+        "!@# tlsn-js {}: {}",
+        phase as u8,
+        phase.get_message().unwrap()
+    );
+}
+
+#[wasm_bindgen]
+pub async fn prover(
+    target_url_str: &str,
+    val: JsValue,
+    secret_headers: JsValue,
+    secret_body: JsValue,
+) -> Result<String, JsValue> {
+    log!("target_url: {}", target_url_str);
+    let target_url = Url::parse(target_url_str)
+        .map_err(|e| JsValue::from_str(&format!("Could not parse target_url: {:?}", e)))?;
+
+    log!(
+        "target_url.host: {}",
+        target_url
+            .host()
+            .ok_or(JsValue::from_str("Could not get target host"))?
+    );
+    let options: RequestOptions = serde_wasm_bindgen::from_value(val)
+        .map_err(|e| JsValue::from_str(&format!("Could not deserialize options: {:?}", e)))?;
+    log!("options.notary_url: {}", options.notary_url.as_str());
+
+    // let fmt_layer = tracing_subscriber::fmt::layer()
+    // .with_ansi(false) // Only partially supported across browsers
+    // .with_timer(UtcTime::rfc_3339()) // std::time is not available in browsers
+    // .with_writer(MakeConsoleWriter); // write events to the console
+    // let perf_layer = performance_layer()
+    //     .with_details_from_fields(Pretty::default());
+
+    // tracing_subscriber::registry()
+    //     .with(tracing_subscriber::filter::LevelFilter::DEBUG)
+    //     .with(fmt_layer)
+    //     .with(perf_layer)
+    //     .init(); // Install these as subscribers to tracing events
+
+    // https://github.com/rustwasm/console_error_panic_hook
```

Comment:
Is it important to keep this? Usually, I prefer to remove out-commented code as it is in git anyways.

Reply:
I agree. But we'll probably enable this again soon.